Bullets and new remoooving structure

Timoxa511 opened the pull request.

Timoxa511 committed:
долго не комител

я всегда подолгу не делаю комиты. я уже добавил пули в игру, и там много
нового. а сейчас я буду менять систему удаления героев

Timoxa511 committed:
я всделал новую систему удаления обьектов

и впервый раз не так долго

ded32 reviewed (changes requested):
I expected you to write the humanistic game, but you made another kill-them-all nightmare... Please add more humanity! :)

Timoxa511 committed:
делаю проверки всего и вся

дампы в начале и конце каждой функции, печать всего и вся. Нашел ошибку.
Нужно доделать проверки еще и в движке и функциях не членов классов

Timoxa511 committed:
все починено

ошибка-глюк-троюродный кот, что был раньше исчез, и все его выходящие
глюки исчезли. Спустя оч много махинаций. А ышо из-за того что я
копепаздил, работа увеличилась на час
#define DA true
#define NET true..............

Timoxa511 commented:
усе готово, надо слить ветки
